content/python/suspend_virtual_server.py.md

```diff
@@ -0,0 +1,364 @@
+}
+
+try:
+    # Order a virtual guest Suspend.
+    response = client['SoftLayer_Product_Order'].placeOrder(orderData)
+    print(json.dumps(response, sort_keys=False, indent=4, separators=(',', ': ')))
+
+except SoftLayer.SoftLayerAPIError as e:
+    """ 
+        If there was an error returned from the SoftLayer API then bomb out with the 
+        error message. 
+        """
+    print("Unable to order a VSI \nfaultCode= %s, \nfaultString= %s"
+          % (e.faultCode, e.faultString))
+
+```
+
+###Check if a VM is Suspended billing
+```python
+import SoftLayer
+
+# Declare the API client
+client = SoftLayer.create_client_from_env()
+
+virtualGuestId = 11111
+
+package_keyName = "SUSPEND_CLOUD_SERVER"
+
+object_mask = 'mask[blockDeviceTemplateGroup,billingItem[package]]'
+
+try:
+    response = client['SoftLayer_Virtual_Guest'].getObject(mask=object_mask, id=virtualGuestId)
+
+    if response['billingItem']['package']['keyName'] == package_keyName:
+        print("The virtual server is a " + str(package_keyName))
+    else:
+        print("The virtual server is not a " + str(package_keyName))
+
+except SoftLayer.SoftLayerAPIError as e:
+    """ 
+        If there was an error returned from the SoftLayer API then bomb out with the 
+        error message. 
+        """
+    print("Unable to get the virtual server information \nfaultCode= %s, \nfaultString= %s"
+          % (e.faultCode, e.faultString))
+
+```
+
+###Check status of VM (suspended or not)
+
+```python
+import SoftLayer
+
+# Declare the API client
+client = SoftLayer.create_client_from_env()
+
+virtualGuestId = 11111
+
+try:
+    # Order a virtual guest Suspended.
+    response = client['SoftLayer_Virtual_Guest'].getPowerState(id=virtualGuestId)
+
+    if response['keyName'] == "HALTED":
+        print("The suspend virtual server is SUSPENDED")
+    else:
+        print("The suspend virtual server is not SUSPENDED")
+
+except SoftLayer.SoftLayerAPIError as e:
+    """ 
+        If there was an error returned from the SoftLayer API then bomb out with the 
+        error message. 
+        """
+    print("Unable to get the virtual server power state information \nfaultCode= %s, \nfaultString= %s"
+          % (e.faultCode, e.faultString))
+
+```
+
+###Get cost related to a Suspended VM
+```python
+import SoftLayer
+from prettytable import PrettyTable
+
+# Declare the API client
+client = SoftLayer.create_client_from_env()
+
+virtualGuestId = 11111
+
+object_mask = 'mask[hoursUsed,hourlyRecurringFee,currentHourlyCharge,invoiceItems[recurringFee,hourlyRecurringFee,' \
+              'createDate]]'
+
+try:
+    billing_response = client['SoftLayer_Virtual_Guest'].getBillingItem(mask=object_mask, id=virtualGuestId)
+
+    state_response = client['SoftLayer_Virtual_Guest'].getPowerState(id=virtualGuestId)
+    
+    # Get the cost related to a suspended virtual server.
+    # Virtual Server is suspended, when it is powerOff. 
+    if state_response['keyName'] == "HALTED":
```

Comment:
Would be good to print out the table on either case here, but if the VSI is not halted, just mention that billing won't include recent usage or something like that.

Reply:
Done